### PR TITLE
Release/0.3.1

### PR DESCRIPTION
**Pull request by lukeindykiewicz**
Update referers.yml (#75) - updated 
Add identified mobile app referers (#79) - cherry picked from https://github.com/snowplow-referer-parser/referer-parser/pull/207
Backport Travis support to 0.3.x (#80) - backport Travis to be able to build `0.3.1` artifact
Drop support for scala 2.9 in 0.3.x branch (#81) - `2.9.1` caused compilation errors

The `maintenance-0.3.x` is going to be "master" for patching `0.3.0`.

### CODE REVIEW

**Review by chuwy — APPROVED** [review]
Just missing final `Prepared for release` commit. Otherwise LGTM!

File: project/BuildSettings.scala [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2012-2013 Snowplow Analytics Ltd
+ * Copyright 2012-2019 Snowplow Analytics Ltd
```
Comment (dilyand) [inline project/BuildSettings.scala]: There should be a 1:1:1 relationship between issue tracker, commit messages and `CHANGELOG`. This gives us better traceability in future. 

For example here we'll need:

issue: `Extend copyright to 2019 (#XX)`
commit message: `Extend copyright to 2019 (close #XX)`
CHANGELOG: `Extend copyright to 2019 (#XX)`

(So this will have to be a separate commit that only changes the copyright term.)

File: project/BuildSettings.scala [hunk 2]
```diff
@@ -22,9 +22,9 @@ object BuildSettings {
   // Basic settings for our app
   lazy val basicSettings = Seq[Setting[_]](
     organization  := "com.snowplowanalytics",
-    version       := "0.3.0",
+    version       := "0.3.1",
```
Comment (dilyand) [inline project/BuildSettings.scala]: 1:1:1

File: project/BuildSettings.scala [hunk 3]
```diff
@@ -22,9 +22,9 @@ object BuildSettings {
...
     description   := "Library for extracting marketing attribution data from referer URLs",
-    scalaVersion  := "2.9.1",
+    scalaVersion  := "2.11.1",
```
Comment (dilyand) [inline project/BuildSettings.scala]: 1:1:1

File: src/main/java/com/snowplowanalytics/refererparser/Medium.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2012-2013 Snowplow Analytics Ltd
+ * Copyright 2012-2019 Snowplow Analytics Ltd
```
Comment (dilyand) [inline src/main/java/com/snowplowanalytics/refererparser/Medium.java]: This should be included in the `Extend copyright...` commit.

File: src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2012-2013 Snowplow Analytics Ltd
+ * Copyright 2012-2019 Snowplow Analytics Ltd
```
Comment (dilyand) [inline src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala]: Copyright commit.

File: src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala [hunk 6]
```diff
@@ -39,6 +39,7 @@ object Medium extends Enumeration {
   val Internal = Value("internal")
   val Social   = Value("social")
   val Email    = Value("email")
+  val Paid     = Value("paid")
```
Comment (dilyand) [inline src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala]: Adding 'Paid' sources might be spun off in its own commit, which is independent from, and a prerequisite for, updating `referers.yml`. Up to you to decide if that would be 'cleaner'.

File: CHANGELOG [hunk 7]
```diff
@@ -1,3 +1,7 @@
+Version 0.3.1 (2019-09-26)
```
Comment (dilyand) [inline CHANGELOG]: Updates to the `CHANGELOG` should be done in a separate `Prepare for release` commit, which will be the only commit that does not close a corresponding issue in the issue tracker and is not noted in the `CHANGELOG`.

File: CHANGELOG [hunk 8]
```diff
@@ -1,3 +1,7 @@
...
+--------------------------
+Update referers.yml to latest version (#75)
```
Comment (dilyand) [inline CHANGELOG]: There should be one entry for each commit; and that should correspond to one issue in the issue tracker.

File: CHANGELOG [hunk 9]
```diff
@@ -1,6 +1,8 @@
 Version 0.3.1 (2019-09-26)
 --------------------------
-Update referers.yml to latest version (#75)
+Update referers.yml (#75)
```
Comment (dilyand) [inline CHANGELOG]: These updates to the `CHANGELOG` should all be done in the `Prepare for release` commit.

File: CHANGELOG [hunk 10]
```diff
@@ -3,6 +3,7 @@ Version 0.3.1 (2019-09-26)
 Update referers.yml (#75)
 Add identified mobile app referers (#79)
 Backport Travis support to 0.3.x (#80)
+Drop support for scala 2.9 in 0.3.x branch (#81)
```
Comment (dilyand) [inline CHANGELOG]: `Prepare for release`

**Review by dilyand — CHANGES_REQUESTED** [review]
Great job @lukeindykiewicz ! I have left some comments to help you restructure the commit history, so that we end up with a 1:1:1 relationship between the issue tracker, commit messages and `CHANGELOG`. This makes it easier to trace the evolution of specific aspects in the code.

You will probably need to open new issues for some commits, to maintain the 1:1:1 relationship. However, those issues might already exist and they might have been fixed in later versions. I think in those cases the way to go will be to have an issue solving the problem in this specific version, eg: `Extend copyright to 2019 in 0.3.1`. This way we can keep the `0.3.1` history separate from the "main" history, while still retaining traceability. (And it shouldn't be a conflict, because newer versions are based not on `0.3.1` but on `0.3.0`, so eg, the copyright would not have been extended in that timeline.)

File: project/BuildSettings.scala (hunk 2 above)
Comment (chuwy) [reply project/BuildSettings.scala]: For non-monorepos we usually bump versions in `Prepared for release` commit

File: src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala (hunk 6 above)
Comment (lukeindykiewicz) [reply src/main/scala/com/snowplowanalytics/refererparser/scala/Parser.scala]: I think one commit for this is enough.

**Comment by lukeindykiewicz** [comment]
Thank you @chuwy and @dilyand for reviews, I tried to stick to all the naming rules. I hope this time it sticks to all the rules :) This was completely redone, so please look at the code again.

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+jdk:
+- openjdk8
+script:
+- sbt test
+deploy:
+  skip_cleanup: true
+  provider: script
+  script: "./.travis/deploy.sh $TRAVIS_TAG"
+  on:
+    condition: '"${TRAVIS_SCALA_VERSION}" == "2.11.1"'
```
Comment (benjben) [inline .travis.yml]: Why do we keep `2.10.4` above if the condition is only met for `2.11.1` ?

**Review by benjben — APPROVED** [review]
LGTM!

File: .travis.yml (hunk 11 above)
Comment (lukeindykiewicz) [reply .travis.yml]: Because this only triggers deploy script, which runs `+publish`, so it publishes to all versions. This is mainly copy-pasted from `0.4.0`.

**Review by dilyand — APPROVED** [review]
👍🏻
